src/preprocessing/passes/non_clausal_simp.cpp

```diff
@@ -448,10 +438,154 @@ PreprocessingPassResult NonClausalSimp::applyInternal(
         Rewriter::rewrite(Node(learnedBuilder)));
   }
 
-  propagator->setNeedsFinish(true);
   return PreprocessingPassResult::NO_CONFLICT;
 }  // namespace passes
 
+SatValue NonClausalSimp::solveByCryptominisat(
+    CVC4::prop::TseitinCnfStream* d_cnfStream,
+    AssertionPipeline* assertionsToPreprocess,
+    unsigned substs_index)
+{
+  for (size_t i = 0, size = assertionsToPreprocess->size(); i < size; ++i)
+  {
+    Trace("non-clausal-simplify") << "Assertion #" << i << " : "
+                                  << (*assertionsToPreprocess)[i] << std::endl;
+  }
```

Comment:
It looks like this tracing stuff can be moved from the two `solve_*` methods to `applyInternal` because it just prints out the assertions that we are preprocessing.